libyul/optimiser/VarNameCleaner.h

```diff
@@ -0,0 +1,82 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+
+#include <libyul/AsmDataForward.h>
+#include <libyul/optimiser/ASTWalker.h>
+
+#include <boost/optional.hpp>
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace yul
+{
+
+struct Dialect;
+
+/**
+ * Pass to clean variable names with hard-to-read names, that happened due to
```

Comment:
I think this should be more specific - it does not improve all names that are hard to read. It strips decimal suffixes after underscore and re-assigns those starting from 1 in case of collisions.

It should also reset the "used map" for every function.